analysis/estimate/notebooks/compare-estimate-consensus.Rmd

```diff
@@ -0,0 +1,354 @@
+  aes(x = estimate_score_immune, y = celltype_immune) + 
+  geom_point() + 
+  geom_smooth(method = "lm") +
+  facet_wrap(vars(project_id), scales = "free_y", nrow = 1) +
+  labs(
+    title = "Immune celltypes", 
+    x = "ESTIMATE immune score (pseudobulk)", 
+    y = "Immune cell fraction\nin consensus celltypes"
+  )
+```
+
+
+```{r, fig.height = 3, fig.width=12}
+ggplot(combined_wide_df) + 
+  aes(x = estimate_score_stromal, y = celltype_stromal) + 
+  geom_point() + 
+  geom_smooth(method = "lm") +
+  facet_wrap(vars(project_id), scales = "free_y", nrow = 1) +
+  labs(
+    title = "Stromal celltypes", 
+    x = "ESTIMATE stromal score (pseudobulk)", 
+    y = "Stromal cell fraction\nin consensus celltypes"
+  )
+```
+
+* Immune plots:
+  * SCPCP000001, SCPCP00006, and SCPCP000017 show a positive relationship between quantities
+  * There is no apparent relationship for SCPCP000002 or SCPCP000009
+* Stromal plots:
+  * SCPCP00006 shows a positive relationship between quantities, and SCPCP000017 _may_ be trending towards a significant positive relationship (but that's a fairly wide +/- SE)
+  * There is no apparent relationship for SCPCP000001, SCPCP000002, or SCPCP000009
+
+That said, SCPCP000006 has an outlier point that may be driving the observed positive relationships.
+
+### Relationship between other cell types and tumor purity
+
+In theory, the `Unknown` cells in our consensus annotation are more likely to be tumor cells.
+We also excluded several cell types from analysis which might also be tumor.
+Similarly in theory, the ESTIMATE score (sum of immune and stromal) is expected to be proportional to tumor purity.
+
+Is there a relationship between these quantities?
+
+```{r, fig.height = 3, fig.width=12, message = FALSE}
+unknown_df <- celltype_data |>
+  purrr::map(
+    \(cell_df) {
+        cell_df |>
+          # group to keep these columns
+          dplyr::group_by(sample_id, project_id) |>
+          dplyr::summarize(
+            is_unknown = sum(consensus_annotation %in% other_celltypes), 
+            frac_unknown = is_unknown / dplyr::n()
+          )
+    }
+  ) |>
+  purrr::list_rbind(names_to = "sample_id") |>
+  dplyr::inner_join(combined_wide_df) |>
+  dplyr::mutate(estimate_score = estimate_score_stromal + estimate_score_immune)
+
+ggplot(unknown_df) + 
+  aes(x = estimate_score, y = frac_unknown) + 
+  geom_point() + 
+  geom_smooth(method = "lm") +
+  facet_wrap(vars(project_id), nrow = 1) +
+  labs(
+    x = "ESTIMATE score (pseudobulk)", 
+    y = "Fraction of non-immune/non-stromal\ncell types"
+  )
+```
+There appears to be a negative relationship for SCPCP000001 and SCPCP000017, a very weak negative relationship for SCPCP000006 (possibly outlier driven?), and no relationship for SCPCP000002 or SCPCP000009.
+
+## Relationship between cell types and bulk ESTIMATE scores
+
+We'll do the same comparisons here but with ESTIMATE scores estimated from bulk counts (not bulk TPM) instead of from pseudobulk.
```

Comment:
I don't necessarily expect these to be correlated. If our hypothesis is that we get different cell compositions between bulk and single-cell, then it's likely that these plots will look different than the ones comparing ESTIMATE to pseudobulk.